modules/apps/headless-apio/structured-content/structured-content-apio-impl/src/main/java/com/liferay/structured/content/apio/internal/architect/resource/StructuredContentNestedCollectionResource.java

```diff
@@ -839,8 +844,25 @@ public StructuredContentField(
 
 		public String getDDMFormFieldDataType() {
 			try {
-				return _ddmStructure.getFieldDataType(
+				String dataType = _ddmStructure.getFieldDataType(
 					_ddmFormFieldValue.getName());
+
+				String displayDataType = dataType;
+
+				if (Objects.equals(dataType, "document-library")) {
+					displayDataType = "document";
+				}
+				else if (Objects.equals(dataType, "journal-article")) {
```

Comment:
Any reason for using Objects.equals in particular?

Reply:
Not needed in this case. Changed!